### DISCUSSION TITLE
Warning building from source as the bottle needs

### DISCUSSION THREAD

**Discussion by jaraco** [Homebrew/discussions#272] [Everyday usage] [answered]
Recently, I ran `brew upgrade` in a macOS 10.15 environment with a non-standard prefix:

```
~ $ brew --prefix
/Users/jaraco/.local/homebrew
```

But when I did, it emitted a lot of warnings:

```
==> Upgrading python@3.8 3.8.6 -> 3.8.6_2 
Warning: Building openssl@1.1 from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /Users/jaraco/.local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /Users/jaraco/.local/homebrew)
- HOMEBREW_REPOSITORY: /usr/local/Homebrew (yours is /Users/jaraco/.local/homebrew)
==> Downloading https://www.openssl.org/source/openssl-1.1.1i.tar.gz
######################################################################## 100.0%
Warning: Building sqlite from source as the bottle needs:
- HOMEBREW_CELLAR: any (yours is /Users/jaraco/.local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /Users/jaraco/.local/homebrew)
- HOMEBREW_REPOSITORY: /usr/local/Homebrew (yours is /Users/jaraco/.local/homebrew)
==> Downloading https://www.sqlite.org/2020/sqlite-autoconf-3340000.tar.gz
######################################################################## 100.0%
Warning: Building python@3.8 from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /Users/jaraco/.local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /Users/jaraco/.local/homebrew)
- HOMEBREW_REPOSITORY: /usr/local/Homebrew (yours is /Users/jaraco/.local/homebrew)
==> Downloading https://files.pythonhosted.org/packages/a7/e0/30642b9c2df516506d40b563b0cbd080c49c6b3f11a70b4c7a670f13a78b/setuptools-50.3.2.zip
######################################################################## 100.0%
==> Downloading https://files.pythonhosted.org/packages/0b/f5/be8e741434a4bf4ce5dbc235aa28ed0666178ea8986ddc10d035023744e6/pip-20.2.4.tar.gz
######################################################################## 100.0%
==> Downloading https://files.pythonhosted.org/packages/83/72/611c121b6bd15479cb62f1a425b2e3372e121b324228df28e64cc28b01c2/wheel-0.35.1.tar.gz
Already downloaded: /Users/jaraco/Library/Caches/Homebrew/downloads/1aa1a9563d6af09f2712c180679dfd512aac6c710fc3fb573175e88eaa7c6de3--wheel-0.35.1.tar.gz
==> Downloading https://www.python.org/ftp/python/3.8.6/Python-3.8.6.tar.xz
Already downloaded: /Users/jaraco/Library/Caches/Homebrew/downloads/8cded16fbf5132bcdeb3125515c30aa484aeb4bdaf7b6886cefcbf129fd47d3d--Python-3.8.6.tar.xz
==> Installing dependencies for python@3.8: openssl@1.1 and sqlite
==> Installing python@3.8 dependency: openssl@1.1
...
```

I see [the docs](https://docs.brew.sh/Installation) advise to _pick another prefix at your peril_, but unfortunately, I'm unable to install to `/usr/local`.

Of course, installing from source is slow and tedious, but still valuable.

I seem to recall this limitation wasn't the case prior to the updates for Big Sur on ARM; am I wrong? Is it truly the case that for Homebrew to support macOS 11 on ARM that the experience for macOS 10 on x86 must be degraded?

**Comment by SMillerDev** [comment, marked as answer]
> I seem to recall this limitation wasn't the case prior to the updates for Big Sur on ARM; am I wrong? 

It was always there, maybe the detection of conflicts/different locations is just a little better now that we set up for ARM.

> Is it truly the case that for Homebrew to support macOS 11 on ARM that the experience for macOS 10 on x86 must be degraded?

We have no intention of degrading usability for any supported configuration, no matter the architecture. Your configuration isn't supported though.

**Reply by jaraco** [reply]
Sounds good. Thanks for the courteous answer.

**Reply by Rylan12** [reply]
I think the change that has been made was only for cases when `HOMEBREW_REPOSITORY == HOMEBREW_PREFIX` which appears to be true in your case. If, instead, you can set up the `HOMEBREW_REPOSITORY` to be in e.g. `/Users/jaraco/.local/homebrew/Homebrew` that _might_ allow this particular bottle to continue to work.

That's just a total guess, though.

**Reply by azrst** [reply]
@Rylan12 Hi Rylan is that just only change name folder? is that any impact for another installed app from brew?
